src/types/mixins.js

```diff
@@ -0,0 +1,25 @@
+'use strict';
+import {isString} from '../validators';
+
+/**
+ * AttributeMixin
+ * @description This is a mixin for adding attribute support to a node
+ * Attributes are simple key:value pairs meant to store any client specific implementation details for a node
+ * @mixin
+ */
+export const AttributeMixin = Object.create(null, {
```

Comment:
Let's chat about the need for this mixin when you're in the office to make sure it's not overkill.